src/main/scala/scalaz/stream/Process.scala

```diff
@@ -451,8 +451,19 @@ sealed trait Process[+F[_], +O]
 
   }
 
+  final def uncons[F2[x] >: F[x], O2 >: O](implicit F: Monad[F2], C: Catchable[F2]): F2[(O2, Process[F2, O2])] =
+    unconsOption(F, C).map(_.getOrElse(throw new NoSuchElementException))
```

Comment:
I would rather this be `flatMap(_.map(_.point).getOrElse(C.fail(new NoSuchElementException)))`, rather than relying on magical exception catching behavior in certain implementations.